src/types/custom.ts

```diff
@@ -0,0 +1,20 @@
+import { Runtype, Rt, Static, create, validationError } from '../runtype'
+import { String } from './string'
+
+export interface Custom<A extends Rt, T extends string, K> extends Runtype<Static<A>> {
+  tag: T,
+  underlying: A,
+  args: K
+}
+
+export function Custom<A extends Rt, T extends string, K>(underlying: A, constraint: (x: Static<A>, args: K) => boolean | string, tag: T, args: K) {
```

Comment:
You can write that as

```ts
CustomArray: (args => Custom(Array(Number), (x, args) => x.length > args.min, 'Length', args))({min:3})
```

if you really need to write it inline like that without repeating yourself.